modules/nlu/src/backend/engine.ts

```diff
@@ -356,9 +365,47 @@ export default class ScopedEngine implements Engine {
         const trainableIntents = intentDefs.filter(i => (i.utterances[lang] || []).length >= MIN_NB_UTTERANCES)
 
         if (trainableIntents.length) {
-          const ctx_intent_models = await this.intentClassifiers[lang].train(trainableIntents, modelHash)
+          const intentsWTokens = await Promise.all(
+            trainableIntents.map(getIntentsWithTokens(lang, this.languageProvider))
+          )
+
+          const tfIdf = computeTfidf(intentsWTokens)
+          const token2vec = {}
+
+          /// this should rather be token2vec = comuteToken2Vec but it was too much rewrite
+          await computeToken2Vec(token2vec, intentsWTokens, lang, this.languageProvider)
+
+          const tfIdfModel = {
+            meta: { context: 'all', created_on: Date.now(), hash: modelHash, scope: 'bot', type: 'intent-tfidf' },
+            model: new Buffer(
+              JSON.stringify({
+                token2vec,
+                l0Tfidf: tfIdf['l0']['__avg__'],
+                l1Tfidf: _.chain(tfIdf)
+                  .map((tfidf, ctx) => ({ tfidf, ctx }))
+                  .reject(model => model.ctx === 'l0')
+                  .map(model => ({ [model.ctx]: model.tfidf['__avg__'] }))
```

Comment:
Isn't it weird to map a dictionnary to an object just to reject l0?

maybe it could be a little sexier like this:

```
_.chain(tfIdf)
.pickBy((tfidf, ctx) => ctx !== 'l0')
.mapValues(tfidf => tfid['__avg__'])
.reduce((a, b) => ({ ...a, ...b }), {})
.value()
```

+ what is the goal of this line?

> .reduce((a, b) => ({ ...a, ...b }), {})

I'm curious on it's output. Does b includes both key and value, or is it just the value? In this case, what is the result of spreading a number? Have you tested the ouput?